### DISCUSSION TITLE
Removing Items from ListView won't refresh

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [tui-cs/Terminal.Gui#2887] [Q&A]
I am stumped and tired of banging my head against the wall. I have a list view that uses an external source. I have code that can add an item to the external source, refresh the list view, and it properly displays the update list. But when I run code to remove an item from the external source, _*using the same list view code*_ won't refresh the list in the app. I have debugged and verified the item is removed in the external source, and that the list source also reflects the change. I simply can't get the listview to refresh and display the updated source. Again, adding an item works fine, but using the same code to remove an item doesn't refresh the list.

Maybe I'm just accidentally lucky getting the added item to show and I'm missing a specific step.

**Comment by BDisp** [comment]
Try `listView.SetNeedsDisplay()` after removed the item.

**Reply by tznind** [reply]
@jdhitsolutions Is there any async code going on? if your external source has `await` or async methods you would need to use the [invoke system](https://gui-cs.github.io/Terminal.Gui/articles/mainloop.html?q=main%20loop)

**Comment by jdhitsolutions** [comment]
I was setting the control to refresh. I think I need a break before I shoot myself in the foot anymore. I created my own PowerShell problem by not paying attention to scope. I had a listview in my dialog with the same name as the listview in the main form. When I called the update function, it was updating the dialog listview and not the main listview. I thought I was being clever but instead blew my big toe off.

**Reply by tig** [reply]
> I thought I was being clever but instead blew my big toe off.

Been there. Done that. Got the tee shirt. 

Glad you figure it out.
